test/core/derivatives/implementation/backpack.py

```diff
@@ -1,65 +1,108 @@
+        """Initialization.
+
+        Args:
+            problem: test problem
+        """
         problem.extend()
         super().__init__(problem)
 
     def store_forward_io(self):
+        """Do one forward pass.
+
+        This implicitly saves relevant quantities for backward pass.
+        """
         self.problem.forward_pass()
 
-    def jac_mat_prod(self, mat):
+    def jac_mat_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.jac_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def jac_t_mat_prod(self, mat):
+    def jac_t_mat_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.jac_t_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def weight_jac_t_mat_prod(self, mat, sum_batch):
+    def weight_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.weight_jac_t_mat_prod(
             self.problem.module, None, None, mat, sum_batch=sum_batch
         )
 
-    def bias_jac_t_mat_prod(self, mat, sum_batch):
+    def bias_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.bias_jac_t_mat_prod(
             self.problem.module, None, None, mat, sum_batch=sum_batch
         )
 
-    def weight_jac_mat_prod(self, mat):
+    def weight_jac_mat_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.weight_jac_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def bias_jac_mat_prod(self, mat):
+    def bias_jac_mat_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.bias_jac_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def ea_jac_t_mat_jac_prod(self, mat):
+    def bias_ih_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.bias_ih_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def bias_hh_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.bias_hh_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def weight_ih_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.weight_ih_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def weight_hh_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.weight_hh_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
```

Comment:
Refactor the body of all `param_jac_t`s into a documented `_param_jac_t_mat_prod(self, mat, sum_batch, param_str)` to reduce duplication.

Reply:
At the moment, all the methods are documented in the abstract class ``DerivativesImplementation`` in base.py.
Thus, these methods are implemented in ``AutogradDerivatives``but don't need a new docstring.
Sadly, I did not find a better solution than just adding ``# noqa: D102``. In my case, the docstring is correctly copied from the parent class, but pydocstyle doesn't recognize it.

Do you still think a refactorization is the right way?

Reply:
For completeness, I found a similar question on the `pydocstyle` repo (https://github.com/PyCQA/pydocstyle/issues/274).

Reply:
Such a refactoring seems necessary for me in the future, but I understand that this PR is mainly concerned with the `RNN` partial derivative implementation. Can you open an issue on `backpack-discuss` to document this task for the future?

Reply:
Created the issue: https://github.com/fKunstner/backpack-discuss/issues/101